server/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/status/TransportSnapshotsStatusAction.java

```diff
@@ -141,7 +141,7 @@ protected void masterOperation(
             client.executeLocally(
                 TransportNodesSnapshotsStatus.TYPE,
                 new TransportNodesSnapshotsStatus.Request(nodesIds.toArray(Strings.EMPTY_ARRAY)).snapshots(snapshots)
-                    .timeout(request.masterNodeTimeout()),
+                    .timeout(request.masterNodeTimeout().millis() < 0 ? null : request.masterNodeTimeout()),
```

Comment:
Shouldn't we compare with -1 explicitly here and everywhere down below? 
Since user can specify any negative number in theory. Or doc should say that any negative number works as infinite  timeout (which sounds confusing)

Reply:
We only accept `-1` when parsing a time value:

https://github.com/elastic/elasticsearch/blob/0699c9351f1439e246d408fd6538deafde4087b6/libs/core/src/main/java/org/elasticsearch/core/TimeValue.java#L375-L384

However the convention elsewhere in the code seems to be to compare `.millis() < 0`. I think we could make this neater for sure, but that's a job for another day. The docs definitely don't need to mention any lenience in this area.